core/trino-main/src/main/java/io/trino/connector/DynamicCatalogManagerModule.java

```diff
@@ -35,13 +37,15 @@ protected void setup(Binder binder)
         if (buildConfigObject(ServerConfig.class).isCoordinator()) {
             binder.bind(CoordinatorDynamicCatalogManager.class).in(Scopes.SINGLETON);
             CatalogStoreConfig config = buildConfigObject(CatalogStoreConfig.class);
-            switch (config.getCatalogStoreKind().toLowerCase(Locale.ROOT)) {
+            String catalogStoreKind = config.getCatalogStoreKind().toLowerCase(Locale.ROOT);
+            switch (catalogStoreKind) {
                 case "memory" -> binder.bind(CatalogStore.class).to(InMemoryCatalogStore.class).in(Scopes.SINGLETON);
                 case "file" -> {
                     configBinder(binder).bindConfig(FileCatalogStoreConfig.class);
                     binder.bind(CatalogStore.class).to(FileCatalogStore.class).in(Scopes.SINGLETON);
                 }
                 default -> {
+                    newOptionalBinder(binder, Key.get(String.class, CatalogStoreKind.class)).setBinding().toInstance(catalogStoreKind);
```

Comment:
I don't think the optional binder is needed (is more complex).  CatalogStoreManager can just have a dependency on CatalogStoreConfig.

Reply:
🤦 I was thinking there must be a simpler way but was tunnel-visioned at the time.

Reply:
Resolved.